lib/doctor/Doctor.ts

```diff
@@ -0,0 +1,314 @@
+    }
+
+    this.diagnosis?.addCliSection({
+      type: 'success',
+      message: i18n(`${localI18nKey}.installed`, {
+        npmVersion,
+      }),
+    });
+  }
+
+  private async checkIfNpmInstallRequired() {
+    let foundError = false;
+    const localI18nKey = `${i18nKey}.projectDependenciesChecks`;
+
+    for (const packageFile of this.diagnosticInfo?.packageFiles || []) {
+      const packageDirName = path.dirname(packageFile);
+      try {
+        const needsInstall = await packagesNeedInstalled(
+          path.join(this.projectConfig?.projectDir, packageDirName)
+        );
+
+        if (needsInstall) {
+          foundError = true;
+
+          this.diagnosis?.addProjectSection({
+            type: 'warning',
+            message: i18n(`${localI18nKey}.missingDependencies`, {
+              dir: packageDirName,
+            }),
+            secondaryMessaging: i18n(
+              `${localI18nKey}.missingDependenciesSecondary`,
+              {
+                command: uiCommandReference('hs project install-deps'),
+              }
+            ),
+          });
+        }
+      } catch (e) {
+        foundError = true;
+
+        if (!(await this.isValidJsonFile(packageFile))) {
+          this.diagnosis?.addProjectSection({
+            type: 'error',
+            message: i18n(`${i18nKey}.files.invalidJson`, {
+              filename: packageFile,
+            }),
+          });
+        } else {
+          this.diagnosis?.addProjectSection({
+            type: 'error',
+            message: i18n(`${localI18nKey}.unableToDetermine`, {
+              dir: packageDirName,
+            }),
+          });
+        }
+
+        logger.debug(e);
+      }
+    }
+
+    if (!foundError) {
+      this.diagnosis?.addProjectSection({
+        type: 'success',
+        message: i18n(`${localI18nKey}.success`),
+      });
+    }
+  }
+
+  private async isValidJsonFile(filename: string) {
+    try {
+      const readFile = util.promisify(fs.readFile);
+      const fileContents = await readFile(filename);
+      JSON.parse(fileContents.toString());
+    } catch (e) {
+      return false;
+    }
+    return true;
+  }
+
+  private async checkProjectConfigJsonFiles() {
+    let foundError = false;
+    for (const jsonFile of this.diagnosticInfo?.configFiles || []) {
+      const fileToCheck = path.join(this.projectConfig?.projectDir, jsonFile);
+      if (!(await this.isValidJsonFile(fileToCheck))) {
+        foundError = true;
+        this.diagnosis?.addProjectSection({
+          type: 'error',
+          message: i18n(`${i18nKey}.files.invalidJson`, {
+            filename: jsonFile,
+          }),
+        });
+      }
+    }
+
+    if (!foundError) {
+      this.diagnosis?.addProjectSection({
+        type: 'success',
+        message: i18n(`${i18nKey}.files.validJson`),
+      });
+    }
+  }
+
+  private async checkIfPortsAreAvailable() {
+    const localI18nKey = `${i18nKey}.port`;
+    if (await isPortManagerServerRunning()) {
```

Comment:
Yeah, I agree that this can be augmented to check if the service running is the actual port manager service.  I think that will be a good candidate for the next iteration!

Reply:
Since we had an issue to update the portManager shutdown flow once [this LDL PR is merged](https://github.com/HubSpot/hubspot-local-dev-lib/pull/209), I can update this to check if the port is available, and then check if it is the port manager